File: 0007-reverse-integer/0007-reverse-integer.py
class Solution(object):
    def reverse(self, x):
        r=0
        s=-1 if x<0 else 1
        x=abs(x)
        while x:
            z=x%10
            x//=10
            r=r*10+z
        if -2**31 <= r <= 2**31 - 1:
            return s*r
        else:
            return 0
            
       
                     
            
           
        
        
            



        """
        :type x: int
        :rtype: int
        """
        

File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long r = 0;
        int s = (x < 0) ? -1 : 1;

        long num = abs((long)x);

        while (num) {
            int z = num % 10;
            num /= 10;
            r = r * 10 + z;
        }

        if (r >= INT_MIN && r <= INT_MAX)
            return s * r;

        return 0;
    }
};